src/briefcase/platforms/linux/appimage.py

```diff
@@ -156,52 +157,69 @@ def build_app(self, app: BaseConfig, **kwargs):
         self.logger.info(f"[{app.app_name}] Building AppImage...")
 
         try:
-            # Build the AppImage.
-            # For some reason, the version has to be passed in as an
-            # environment variable, *not* in the configuration...
-            env = {"VERSION": app.version}
-
-            # Find all the .so files in app and app_packages,
-            # so they can be passed in to linuxdeploy to have their
-            # dependencies added to the AppImage. Looks for any .so file
-            # in the application, and make sure it is marked for deployment.
-            so_folders = set()
-            for so_file in self.appdir_path(app).glob("**/*.so"):
-                so_folders.add(so_file.parent)
-
-            deploy_deps_args = []
-            for folder in sorted(so_folders):
-                deploy_deps_args.extend(["--deploy-deps-only", str(folder)])
-
-            # Build the app image. We use `--appimage-extract-and-run`
-            # because AppImages won't run natively inside Docker.
-            with self.dockerize(app) as docker:
-                docker.run(
-                    [
-                        self.linuxdeploy.appimage_path,
-                        "--appimage-extract-and-run",
-                        f"--appdir={self.appdir_path(app)}",
-                        "-d",
-                        os.fsdecode(
-                            self.appdir_path(app)
-                            / f"{app.bundle}.{app.app_name}.desktop"
-                        ),
-                        "-o",
-                        "appimage",
-                    ]
-                    + deploy_deps_args,
-                    env=env,
-                    check=True,
-                    cwd=self.platform_path,
-                )
-
-            # Make the binary executable.
-            self.os.chmod(self.binary_path(app), 0o755)
+            self._build_appimage(app)
         except subprocess.CalledProcessError as e:
             raise BriefcaseCommandError(
                 f"Error while building app {app.app_name}."
             ) from e
 
+    def _build_appimage(self, app):
+        """Build the AppImage.
+
+        :param app: The application to build
+        """
+
+        # For some reason, the version has to be passed in as an
+        # environment variable, *not* in the configuration...
+        env = {"VERSION": app.version}
+        plugins = []
+        if app.requires:
+            for dependency in app.requires:
+                if "toga-gtk" in dependency or "PyGObject" in dependency:
```

Comment:
Will this catch:
* `toga-gtk>=0.3.0.dev33`? (i.e., a version string on the dependency)?
* `pygobject` (i.e., case incompatibility?)
* An app that uses a different toolkit that wraps GTK, but doesn't directly import pygobject? (not sure I have any specific examples in mind, but I'm sure *someone* has built one)
?

On the last example - is there a reason you opted for dependency inspection rather than an explicit list of linuxdeploy plugins needed by the app?

Reply:
>  toga-gtk>=0.3.0.dev33? (i.e., a version string on the dependency)?

Yes, I think so, that is what I tested it with a toga-gtk is in that string.

> pygobject (i.e., case incompatibility?)

Updated the implementation to support this, good idea!

> An app that uses a different toolkit that wraps GTK, but doesn't directly import pygobject? (not sure I have any specific examples in mind, but I'm sure someone has built one)

No, but we could support this in the future if we had something in mind.

> On the last example - is there a reason you opted for dependency inspection rather than an explicit list of linuxdeploy plugins needed by the app?

I think the general use case this was implementing for was users either have a Toga or PyGObject app and those will be listed already in the requires. We could add another option for explicit plugins needed, but I am not sure that this is needed at this time unless we implement a much larger plugin system, and if we do that we could add it at that time.